### DISCUSSION TITLE
status of enzyme

### DISCUSSION THREAD

**Discussion by vavrines** [LuxDL/Lux.jl#406] [General]
It is mentioned in the [documentation](https://lux.csail.mit.edu/dev/tutorials/beginner/1_Basics#automatic-differentiation) that there is a plan to move towards Enzyme.
I would like to learn if there is an estimated time for this migration and what is the current status.
Thanks in advance : )

**Comment by avik-pal** [comment]
There is no timeline for now. But there is some upstream activity to make Enzyme compatible with SciMLSensitivity.jl which by extension will make Lux compatible with Enzyme.

FWIW most simple things already work, but things that remain are the chain rules core rules for LuxLib and NNlib that need to be migrated.

**Comment by avik-pal** [comment]
Enzyme is now the recommended backend for CPU models https://lux.csail.mit.edu/stable/manual/autodiff#autodiff-recommendations. GPU support is underway in Enzyme.jl repo
